client/me/purchases/manage-purchase/index.jsx

```diff
@@ -46,22 +46,33 @@ import {
 	showCreditCardExpiringWarning,
 	showEditPaymentDetails
 } from 'lib/purchases';
-import { getPurchase, goToList, isDataLoading, recordPageView } from '../utils';
+import { getPurchase, getSelectedSite, goToList, isDataLoading, recordPageView } from '../utils';
 import { isDomainProduct, isGoogleApps, isPlan, isSiteRedirect, isTheme } from 'lib/products-values';
 
 const ManagePurchase = React.createClass( {
 	propTypes: {
 		cart: React.PropTypes.object.isRequired,
 		selectedPurchase: React.PropTypes.object.isRequired,
-		selectedSite: React.PropTypes.object,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
 		destinationType: React.PropTypes.string
 	},
 
 	componentWillMount() {
+		if ( ! this.isDataValid() ) {
+			return page.redirect( paths.list() );
```

Comment:
> I think the redirect should happen a bit higher in the stack, preferably the controller.

> I agree that this could be moved one level up to `<StoreConnection />` component.

Agreed. Thanks for pointing this out.